Sources/NIOCore/ByteBuffer-aux.swift

```diff
@@ -125,6 +137,23 @@ extension ByteBuffer {
             return self._setStringSlowpath(string, at: index)
         }
     }
+    
+    /// Write `string` null terminated into this `ByteBuffer` at `index` using UTF-8 encoding. Does not move the writer index.
+    ///
+    /// - parameters:
+    ///     - string: The string to write.
+    ///     - index: The index for the first serialized byte.
+    /// - returns: The number of bytes written.
+    public mutating func setNullTerminatedString(_ string: String, at index: Int) -> Int {
+        // Do not implement setString via setSubstring. As of Swift version 5.1.3,
+        // Substring.UTF8View does not implement withContiguousStorageIfAvailable
+        // and therefore has no fast access to the backing storage.
+        return string.withCString { ptr in
+            let urbp = UnsafeRawBufferPointer(start: .init(ptr), count: strlen(ptr) + 1)
```

Comment:
we could use `string.utf8.count` instead of `strlen(ptr)` but not 100% we should rely on this...

Reply:
```